resources/version.go

```diff
@@ -103,7 +106,7 @@ func CreateVersion(db *gorm.DB, client kube.APIInterface, version *Version) (boo
 
 	// update environment service
 	ports := gatherContainerPorts(version.Containers)
-	UpdateEnvironmentService(db, client, &version.Environment, ports)
+	UpdateEnvironmentService(db, client, version.Environment, ports)
```

Comment:
There was an error regarding recursive type on the Version struct that was related to the Environment field type. To fix this I needed to set the Environment field on the Version struct to a pointer.